### PR TITLE
Add Contributing guidelines

### PR DESCRIPTION
**Pull request by YourFin**
Sorry that this is such a big commit, and I would say that it does break my recommendations on how big to make pull requests. I would ask that you review this document thoroughly up until the "Git tricks" section; and skim the "Git problems, and how to solve them" section. The last bit covers most of the weirdness that I personally have run into with the proposed workflow.

Note that this is a pull request, and as such I am happy to take suggestions and criticism on this document.

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    export VISUAL='nano'
+
+To keep these changes when you re-open your shell, you should add these two lines to your `~/.bashrc`.
+
+To do this on windows, see [this](https://stackoverflow.com/questions/2114111/where-does-git-config-global-get-written-to) entry on stackoverflow.
+
+#### Commit size
+The big thing to keep in mind is that you really don't want to make much progress without putting in a pull request.
+Try to keep each pull request to a single commit, doing a single change.
+As a rough metric; 80 lines of code added is probably too many, as a rough estimate.
```
Comment (evan-w-holt) [inline Contributing.md]: Grammatical thing. "as a rough estimate" is redundant after starting the sentence with "As a rough metric."

File: Contributing.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+When you have some changes that you want reviewed, you should use the github pull request system to get your changes reviewed.
+This can be done by
+1. Opening up the web ui
+1. Clicking the "branch" button on the left hand side
+1. Clicking "Compare and pull request"
+Alternatively, you can look into the official [hub](https://hub.github.com/) command line tool.
+
+#### Pull request messages
+In general these can be short or nonexistent, as most of the information that you would want here is in your commit message.
+What does make sense to put here are things that relevant only at the time that you make the pull request, or messages directed specifically at the code reviewer.
```
Comment (evan-w-holt) [inline Contributing.md]: Add "are" between "that" and "relevant."

File: Contributing.md (hunk 1 above)
Comment (YourFin) [reply Contributing.md]: Thanks! Resolved.

File: Contributing.md (hunk 2 above)
Comment (YourFin) [reply Contributing.md]: Thanks! Suggested changes applied :)

**Pull request merged**